components/builder-depot/src/config.rs

```diff
@@ -224,10 +213,8 @@ mod tests {
         assert_eq!(config.http.port, 9000);
         assert_eq!(&format!("{}", config.routers[0]), "172.18.0.2:9001");
         assert_eq!(config.targets.len(), 2);
-        assert_eq!(config.targets[0].platform, Platform::Linux);
-        assert_eq!(config.targets[0].architecture, Architecture::X86_64);
-        assert_eq!(config.targets[1].platform, Platform::Windows);
-        assert_eq!(config.targets[1].architecture, Architecture::X86_64);
+        assert_eq!(config.targets[0], target::X86_64_LINUX);
+        assert_eq!(config.targets[1], target::X86_64_WINDOWS);
```

Comment:
There is code in builder-depot that does a target check like:
    if !depot.targets.contains(&target_from_artifact) ...

We need to either update that, or add a test here that ensures that targets cannot contain any other known target.

Reply:
Let me double check this code, but as we're comparing a `PackageIdent` with presumably a `Vec<PackageIdent`, the `Eq`/`PartialEq` implementations allow this to keep working. The value at those constants are internally types like `PackageTarget(Type::X86_64_Linux))` and `PackageTarget(Type::X86_64_Windows))` which is the same types in `depot.targets` (which I presume is derived from `Config`).

Anyway, I'll double check

Reply:
I'm seeing 3 call sites with the `depot.target.contains(&target)` pattern in the `server` module:

* https://github.com/habitat-sh/builder/blob/f46ac9e992d5637bacc97ba147f34421054404ce/components/builder-depot/src/server.rs#L819 in `upload_package()`
* https://github.com/habitat-sh/builder/blob/f46ac9e992d5637bacc97ba147f34421054404ce/components/builder-depot/src/server.rs#L1242 in `download_package()`
* https://github.com/habitat-sh/builder/blob/f46ac9e992d5637bacc97ba147f34421054404ce/components/builder-depot/src/server.rs#L2492 in `download_package_from_upstream_depot()`

In all 3 places, it's checking in `depot.targets` which is a read-only borrow of `Config` to get the `Vec<PackageTarget>` vec and uses [`contains`](https://doc.rust-lang.org/std/primitive.slice.html#method.contains) to get a `true`/`false` response. As the underlying type `PackageTarget` hasn't changed (only the internals), this is all good and continues to work 100% as before.

Yay!

Reply:
👍